etl/pypasar/omop/note.py

```diff
@@ -22,15 +21,122 @@ def execute(self):
+                connection.execute(
+                    text(f'''
+                        CREATE OR REPLACE VIEW {omop_schema}.stg__note AS
+                            WITH postop__clindoc AS (
+                                SELECT
+                                    id,
+                                    anon_case_no,
+                                    session_id,
+                                    postop_clindoc_created_datetime,
+                                    postop_clindoc_item_name,
+                                    postop_clindoc_item_description,
+                                    postop_clindoc_value_text,
+                                    ROW_NUMBER() OVER (
+                                        PARTITION BY anon_case_no, session_id, postop_clindoc_created_datetime, postop_clindoc_item_description
+                                        ORDER BY id
+                                    ) AS row_num      --Any rows that are duplicate with show 1,2,...
+                                FROM postop.clindoc
+                            ),
+
+                            -- Filter the table to only include unique value
+                            filtered AS (
+                                SELECT * from postop__clindoc WHERE ROW_NUM = 1
+                            )
+
+                            SELECT
+                                ROW_NUMBER() OVER (ORDER BY clindoc.postop_clindoc_created_datetime, id) AS note_id,
+                                CDM_PER.person_id AS person_id,
+                                CAST(clindoc.postop_clindoc_created_datetime AS date) AS note_date, -- get date only
+                                clindoc.postop_clindoc_created_datetime AS note_datetime,
+                                clindoc.postop_clindoc_item_name AS note_title,
+
+                            -- Handle any null values in note text
+                            CASE
+                                WHEN clindoc.postop_clindoc_value_text IS NOT NULL 
+                                THEN clindoc.postop_clindoc_value_text
+                                ELSE '0'
+                            END AS note_text,
+
+                                CDM_VisitOcc.visit_occurrence_id AS visit_occurrence_id,
+                                CDM_VisitDet.visit_detail_id AS visit_detail_id,
+                                clindoc.postop_clindoc_item_description AS note_source_value
+                            FROM filtered AS clindoc
+                            -- Join tables needed for person_id, visit_occurrence_id, visit_detail_id
+                            JOIN {omop_schema}.person AS CDM_PER
+                                ON clindoc.anon_case_no=CDM_PER.person_source_value
+                            JOIN omop_sqldev_schema.visit_occurrence AS CDM_VisitOcc
+                                ON CDM_PER.person_id=CDM_VisitOcc.person_id
```

Comment:
Please join the **visit_occurrence table** using clindoc.`session_id` = CDM_VisitOcc.`session_id` instead, as the code for converting `session_id` is provided below. This change is necessary since only a single person_id can map to multiple occurrences, rather than just one.
```sql
CAST(LEFT(CAST(visit_occurrence_id AS TEXT), LENGTH(CAST(visit_occurrence_id AS TEXT)) - 2) AS INTEGER) AS session_id
```